.github/workflows/deploy-to-sandbox.yml

```diff
@@ -0,0 +1,170 @@
+name: Deploy to Sandbox Environment
+
+on:
+  push:
+    branches: [ main, develop, DEV-* ]
+  pull_request:
+    branches: [ main ]
+  workflow_dispatch:
+
+env:
+  MODULE_NAME: Taxcloud_Magento2
+  MODULE_PATH: app/code/Taxcloud/Magento2
+
+jobs:
+  test:
+    name: Run Tests
+    runs-on: ubuntu-latest
+    
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v4
+      
+    - name: Set up PHP
+      uses: shivammathur/setup-php@v2
+      with:
+        php-version: '8.1'
+        extensions: soap, dom, curl, libxml, mbstring, zip, pcntl, pdo, sqlite, pdo_sqlite, bcmath, soap, intl, gd, exif, iconv
+        coverage: none
+        
+    - name: Install dependencies
+      run: |
+        composer install --no-dev --optimize-autoloader
+        
+    - name: Run integration tests
+      run: |
+        php Test/Integration/PostalCodeParserTest.php
+        php Test/Integration/RefundTest.php
+        php Test/Integration/ComprehensiveRefundTest.php
+        php Test/Integration/FailureCaseTest.php
+        php Test/Integration/FallbackToMagentoTest.php
+        php Test/Integration/AdobeCommerce248p1CompatibilityTest.php
+
+  deploy:
```

Comment:
Maybe worth making actions to split out some of these things, in case we want a staging/prod deploy or w/e later? Your call though

Reply:
Good call will do!